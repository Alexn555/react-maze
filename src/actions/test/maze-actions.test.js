import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const fetchMap = () => ({
  type: 'FETCH_MAP'
});

describe('quiz async actions', () => {
  afterEach(() => {
    fetchMock.restore()
  });

  it('should send FETCH_MAP action and get correct snapshot', () => {
	const store = mockStore({ map: [], loading: [], errors: [] });
    store.dispatch(fetchMap());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should send FETCH_MAP action and get right actions', () => {
    fetchMock.getOnce('/maze-map', {
      body: { payload: ['something'] },
      headers: { 'content-type': 'application/json' }
    });

    const expectedActions = [
      { type: 'FETCH_MAP' }
    ];

    const store = mockStore({ quizzes: [], loading: [], errors: [] });
	store.dispatch(fetchMap());
    expect(store.getActions()).toEqual(expectedActions);
  });


});




