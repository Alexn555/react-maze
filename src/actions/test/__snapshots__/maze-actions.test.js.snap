// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz async actions should send FETCH_MAP action and get correct snapshot 1`] = `
Array [
  Object {
    "type": "FETCH_MAP",
  },
]
`;
